platform-sdk/docs/core/dynamic-address-book/DynamicAddressBook-PlatformDesign.md

```diff
@@ -0,0 +1,134 @@
+
+Buffer: Only the latest events from each node within the same birthRound need to be buffered.
+
+### Remove Node
+
+When a roster with a removed node becomes the `currentEffectiveRoster` for the `pendingConsensusRound`, the event
+creation algorithm is reset.
+
+### Rotate Signing Key
+
+The `currentEffectiveRoster` contains the key used to sign the event. Rotation of the roster will update the signing
+key to use for signing self events.
+
+### Node Weight Change
+
+Changing network weight in the `currentEffectiveRoster` for the `pendingConsensusRound` should cause the event creation
+algorithm to reset.
+
+## Gossip
+
+### Receiving Events
+
+The `futureEventHorizon` and the `pastEventHorizon` define the thresholds for receiving gossipped Events. Received
+events with `birthRound` outside of this node's Event Horizon will be rejected or discarded.
+
+### Sending Events
+
+A node will store ancient events and send them through gossip to help nodes which are behind.
+
+### Add Node
+
+When transaction handling creates a roster with a new node in it, it becomes the `greatestEffectiveRoster` for
+the `futureEventHorizon` and this node will be able to receive these events.
+
+### Remove Node
+
+When a node is removed from a roster and all rosters containing the node have become ancient, this node will no longer
+receive events from the removed node. If received, they will be discarded.
+
+### Rotate Signing Key
+
+This roster change has no effect on whether or not send or receive events through Gossip.
+
+### Node Weight Change
+
+This roster change has no effect on whether or not send or receive events through Gossip.
+
+## Event Validation
+
+For each event, lookup the roster within the `eventHorizon` that matches the `birthRound` of the event. Events
+outside the `eventHorizon` are not validatable.
+
+## Orphan Buffer
+
+Drop ancient events that have a birthRound less than `minRoundNonAncient`.
+
+## ISS Detection
+
+Use the effective roster for the round in question.
+
+## Consensus Effects
+
+Uses the `currentEffectiveRoster` for the `pendingConsnensRound`.
+
+When a round comes to consensus:
+
+* The `latestConsensusRound` and `latestConsensRoster` can be retrieved from the produced `ConsensusRound`.
+* A new `NonAncientEventWindow` is produced on an output wire.
+* Consensus advances the `pendingConsensusRound` and uses the `currentEffectiveRoster` for that round.
+* The `RosterDiff` between the `currentEffectiveRoster` and the `latestConsensusRoster` is produced on an output wire.
+
+## Signature Gathering for State
+
+Use the effective roster for the relevant round when validating signatures on the round's state.
+
+## Reconnect
+
+Validating state signatures.
+
+Use the effective roster for the relevant round when validating signatures on the round's state.
+
+NOTE: The node must be able to validate 1/2 of the weight with its current effective roster. Since 2/3 of the
```

Comment:
```suggestion
NOTE: The node must be able to validate more than 1/2 of the weight with its current effective roster. Since 2/3 of the
```

Reply:
Didn't Leemon say this past Friday that requiring >2/3 signatures to fully sign a state causes problems?

Reply:
yes, if 1/3 of the network has issues, the state signing would fail to be achieved.

Reply:
I think this is ok for now, but we may need to update this later if we decide the "fully signed" threshold needs to go back to >1/2